Sources/PackageCollections/PackageCollections.swift

```diff
@@ -189,6 +176,59 @@ public struct PackageCollections: PackageCollectionsProtocol {
         }
     }
 
+    // MARK: - Packages
+
+    public func findPackages(
+        _ query: String,
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>? = nil,
+        profile: PackageCollectionsModel.Profile? = nil,
+        callback: @escaping (Result<PackageCollectionsModel.PackageSearchResult, Error>) -> Void
+    ) {
+        let profile = profile ?? .default
+
+        self.storage.collectionsProfiles.listSources(in: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let sources):
+                let identifiers = sources.map { .init(from: $0) }.filter { collections?.contains($0) ?? true }
+                if identifiers.isEmpty {
+                    return callback(.success(PackageCollectionsModel.PackageSearchResult(items: [])))
+                }
+                self.storage.collections.searchPackages(identifiers: identifiers, query: query, callback: callback)
+            }
+        }
+    }
+
+    // MARK: - Package Metadata
+
+    public func getPackageMetadata(_ reference: PackageReference,
+                                   profile: PackageCollectionsModel.Profile? = nil,
+                                   callback: @escaping (Result<PackageCollectionsModel.PackageMetadata, Error>) -> Void) {
+        let profile = profile ?? .default
+
+        self.findPackageByIdentifier(reference.identity, profile: profile) { result in
+            switch result {
+            case .failure(let error):
+                callback(.failure(error))
+            case .success(let packageSearchResult):
+                self.metadataProvider.get(reference: reference) { result in
+                    switch result {
+                    case .failure(let error):
+                        callback(.failure(error))
+                    case .success(let basicMetadata):
+                        callback(.success(PackageCollectionsModel.PackageMetadata(
+                            package: self.mergedPackageMetadata(package: packageSearchResult.package, basicMetadata: basicMetadata),
```

Comment:
finally merge the results

Reply:
Makes sense.  For a non-trivial flow, it might help others to add brief comments (just brief, like the ones here in the PR).

Reply:
Apologies if that's obvious and would be added after this PR is developed more.